proxy/http2/Http2Stream.cc

```diff
@@ -808,7 +806,7 @@ Http2Stream::destroy()
   http_parser_clear(&http_parser);
 
   super::destroy();
-  THREAD_FREE(this, http2StreamAllocator, this_ethread());
+  delete this;
```

Comment:
Why don't we just use `delete` instead of calling `destroy()`? I'm not sure if it's really possible, but calling `destroy()` from the destructor sounds more natural to me.

Reply:
These are all very small iterative changes, so no one's head explodes.
What I'm working towards is all Session and Transaction/Streams API will use 
- `dependent = new Class(parent, init_params)` to sync params
  - which will call `parent->reserve(dependent)` to increment reference
- `parent->release(dependent)` to decrement parent's reference and maybe shutdown or delete the parent.

So after this PR I'll make the `destroy()` private, then later a PR merge it into a destructor.

But I want to normalize the API so that externally we always call `release()` or `do_io_close()` on all these classes because sessions have to do a call out for ssn close hooks before destructing and freeing.

Reply:
Thanks for the clarification. I'm ok with this as long as it's an intermediate shape. I still don't understand why `delete` have to be wrapped into a member function, but I'm going to ask it again on the following PR after the code gets close to the final shape.